crayfish/animation.py

```diff
@@ -101,32 +101,37 @@ def prep_comp(cfg, mr, time):
     w,h = mr.outputSize().width(), mr.outputSize().height()
     dpi = mr.outputDpi()
     c = QgsComposition(mr)
+    #suppose to be always set below
+    dataset = None 
+    if cfg['layer'] and cfg['layer'].currentDataSet():
+        dataset = cfg['layer'].currentDataSet()
     if layoutcfg['type'] == 'file':
-        prepare_composition_from_template(c, cfg['layout']['file'], time)
+        prepare_composition_from_template(c, cfg['layout']['file'], time, dataset)
     else:  # type == 'default'
-        cm = prepare_composition(c, w, h, dpi, time, layoutcfg)
+        cm = prepare_composition(c, w, h, dpi, time, layoutcfg, ds=dataset)
     return c
 
 
 def composition_set_time(c, time, frmt=0):
-    timeItem = c.getComposerItemById("time")
-    if timeItem is not None:
-        if frmt == 0: # hh:mm:ss
-            txt = time_to_string(time)
-        else:  # hh.hhh
-            txt = "%06.3f" % time
-        timeItem.setText(txt)
+    for i in c.items():
+        if isinstance(i, QgsComposerLabel) and i.id() == "time":
+            if frmt == 0:  # hh:mm:ss
+                txt = time_to_string(time)
+            else:  # hh.hhh
+                txt = "%06.3f" % time
+            i.setText(txt)
 
 
-def prepare_composition_from_template(c, template_path, time):
+
+def prepare_composition_from_template(c, template_path, time, ds):
 
     document = QDomDocument()
     document.setContent(open(template_path).read())
     c.loadFromTemplate(document)
 
     c.setPlotStyle(QgsComposition.Print)
 
-    composition_set_time(c, time)
+    composition_set_time(c, time, ds)
```

Comment:
hmm, not sure, composition_set_time 3rd parameter is fmt and not dataset? Maybe some wrong (automatic) merge?